dev-tools/reconfigurator-cli/src/lib.rs

```diff
@@ -162,11 +163,32 @@ impl ReconfiguratorSim {
                 builder
                     .add_omicron_zone_external_ip(zone.id, external_ip)
                     .context("adding omicron zone external IP")?;
+
+                // TODO-completeness: This needs to support dual-stack zones.
+                // See https://github.com/oxidecomputer/omicron/issues/9288 and
+                // related issues.
+                let maybe_ip = if matches!(external_ip.ip(), IpAddr::V4(_)) {
+                    nic.ip_config.ipv4_addr().copied().map(IpAddr::V4)
+                } else {
+                    nic.ip_config.ipv6_addr().copied().map(IpAddr::V6)
+                };
+                let ip = maybe_ip.with_context(|| {
+                    format!(
+                        "Omicron zone has an external and private IP \
```

Comment:
Might be clearer to say "Omicron zone has external and private IP configurations with different IP versions". I had to read this a few times to grok what it was saying.